src/test/java/org/sagebionetworks/bridge/services/AccountServiceTest.java

```diff
@@ -0,0 +1,167 @@
+        verify(mockAccountDao).authenticate(study, signIn);
+    }
+
+    @Test
+    public void reauthenticate() {
+        Study study = Study.create();
+        SignIn signIn = new SignIn.Builder().build();
+        Account account = Account.create();
+        when(mockAccountDao.reauthenticate(study, signIn)).thenReturn(account);
+        
+        Account returnVal = service.reauthenticate(study, signIn);
+        assertEquals(returnVal, account);
+        verify(mockAccountDao).reauthenticate(study, signIn);
+    }
+    
+    @Test
+    public void deleteReauthToken() {
+        service.deleteReauthToken(ACCOUNT_ID);
+        verify(mockAccountDao).deleteReauthToken(ACCOUNT_ID);
+    }
+    
+    @Test
+    public void createAccount() {
+        Study study = Study.create();
+        Account account = Account.create();
+        Consumer<Account> consumer = (oneAccount) -> {};
+        service.createAccount(study, account, consumer);
+        verify(mockAccountDao).createAccount(study, account, consumer);
+    }
+    
+    @Test
+    public void updateAccount() {
+        Account account = Account.create();
+        Consumer<Account> consumer = (oneAccount) -> {};
+        service.updateAccount(account, consumer);
+        verify(mockAccountDao).updateAccount(account, consumer);
+    }
+    
+    @Test
+    public void editAccount() {
+        Consumer<Account> consumer = (oneAccount) -> {};
+        service.editAccount(TEST_STUDY, HEALTH_CODE, consumer);
+        verify(mockAccountDao).editAccount(TEST_STUDY, HEALTH_CODE, consumer);
+    }
+    
+    @Test
+    public void getAccount() {
+        Account account = Account.create();
+        when(mockAccountDao.getAccount(ACCOUNT_ID)).thenReturn(account);
+        
+        Account returnVal = service.getAccount(ACCOUNT_ID);
+        assertEquals(returnVal, account);
+        verify(mockAccountDao).getAccount(ACCOUNT_ID);
+    }
+    
+    @Test
+    public void deleteAccount() {
+        service.deleteAccount(ACCOUNT_ID);
+        verify(mockAccountDao).deleteAccount(ACCOUNT_ID);
+    }
+
+    @Test
+    public void getPagedAccountSummaries() {
+        Study study = Study.create();
+        when(mockAccountDao.getPagedAccountSummaries(study, EMPTY_SEARCH)).thenReturn(mockAccountSummaries);
+        
+        PagedResourceList<AccountSummary> returnVal = service.getPagedAccountSummaries(study, EMPTY_SEARCH);
+        assertEquals(returnVal, mockAccountSummaries);
+        verify(mockAccountDao).getPagedAccountSummaries(study, EMPTY_SEARCH);   
+    }
+
+    @Test
+    public void getHealthCodeForAccount() {
+        Account account = Account.create();
+        account.setHealthCode(HEALTH_CODE);
+        when(mockAccountDao.getAccount(ACCOUNT_ID)).thenReturn(account);
+        
+        String healthCode = service.getHealthCodeForAccount(ACCOUNT_ID);
+        assertEquals(healthCode, HEALTH_CODE);
+        verify(mockAccountDao).getAccount(ACCOUNT_ID);
+    }
```

Comment:
We should also have a unit test for the null case.

Reply:
Added